src/js/components/ServiceForm.js

```diff
@@ -0,0 +1,113 @@
+import {Hooks} from 'PluginSDK';
+import React from 'react';
+
+import FormUtil from '../utils/FormUtil';
+import SchemaForm from './SchemaForm';
+import SchemaFormUtil from '../utils/SchemaFormUtil';
+
+const METHODS_TO_BIND = [
+  'handleFormChange',
+  'validateForm'
+];
+
+class ServiceForm extends SchemaForm {
+  constructor() {
+    super();
+
+    METHODS_TO_BIND.forEach((method) => {
+      this[method] = this[method].bind(this);
+    });
+
+    this.store_listeners = Hooks.applyFilter(
+      'serviceFormStoreListeners', []
+    );
+  }
+
+  componentWillMount() {
+    this.multipleDefinition = this.getNewDefinition();
+    this.props.getTriggerSubmit(this.handleExternalSubmit);
+  }
+
+  componentDidMount() {
+    super.componentDidMount(...arguments);
+    Hooks.doAction('serviceFormMount');
+  }
+
+  handleVariableSecretClick(fieldName, prop) {
+    let propIndex = FormUtil.getPropIndex(fieldName);
+    let variablesDefinition = this.multipleDefinition
+      .environmentVariables.definition;
+
+    if (!variablesDefinition) {
+      return;
+    }
+
+    variablesDefinition.forEach(function (field) {
+      if (!Array.isArray(field)) {
+        return;
+      }
+
+      field.forEach(function (fieldColumn) {
+        let propKey = FormUtil.getPropKey(fieldColumn.name);
+        if (FormUtil.isFieldInstanceOfProp(prop, fieldColumn, propIndex) &&
+          propKey === 'value') {
+          if (fieldColumn.fieldType === 'text') {
+            fieldColumn.fieldType = 'select';
```

Comment:
This bit confuses me - can you write a comment as to why we're changing from `text` to `select` and vise versa on L60?